code/src/components/AllMovies.js

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+
+import Movie from "./Movie";
+
+const AllMovies = ({ userApiKey, allMovies, setAllMovies }) => {
+  const history = useHistory();
+
+  const API_URL = `https://api.themoviedb.org/3/movie/popular?api_key=${userApiKey}&language=en-US&page=1`;
+
+  useEffect(() => {
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.results !== undefined) {
+          setAllMovies(data.results);
+        } else {
+          // pushing to homepage because lack of data is probably no API key
+          history.push("/");
+        }
+      });
+  }, [API_URL, setAllMovies, history]);
+
+  return (
+    <div className="all-movies-container">
```

Comment:
Not really responsive in mobile view. I tried to figure out what makes it bu could not really find the issue.. 
![codereview sarah and ru3](https://user-images.githubusercontent.com/81567468/141451024-cc092c45-45d1-4341-b694-38a4770ed720.PNG)